CMakeLists.txt

```diff
@@ -284,11 +290,11 @@ if (MRDOCS_BUILD_TESTS)
     endif ()
     target_compile_definitions(mrdocs-test PRIVATE -DMRDOCS_TEST_FILES_DIR="${CMAKE_CURRENT_SOURCE_DIR}/test-files")
     add_test(NAME mrdocs-unit-tests COMMAND mrdocs-test --unit=true)
-    add_test(NAME mrdocs-golden-tests COMMAND mrdocs-test --unit=false --action=test "${PROJECT_SOURCE_DIR}/test-files/golden-tests" --addons="${CMAKE_SOURCE_DIR}/share/mrdocs/addons")
+    add_test(NAME mrdocs-golden-tests COMMAND mrdocs-test --unit=false --action=test "${PROJECT_SOURCE_DIR}/test-files/golden-tests" --addons="${CMAKE_SOURCE_DIR}/share/mrdocs/addons" --system-includes="${LIBCXX_DIR}")
```

Comment:
`CLANG_STDLIB_INCLUDE_DIR` was set in line 98 but never used here.

Reply:
I'm very suspicious about this. If you really need `CLANG_STDLIB_INCLUDE_DIR`, it's not supposed to work without it. My intuition is you're not providing CLANG_STDLIB_INCLUDE_DIR and the thing is working in CI because it finds it in the system as usual. 

The main problem related to this blocking us now is we're trying to build documentation in Ubuntu 20.04 because that's what the boost release tools use. To emulate that and be sure the binaries works elsewhere the `website` workflow in ci.yml used to use Ubuntu 20.04 to emulate the conditions in the boost release tools. The reason we have to disable 20.04 and use 23.04 is this very issue.

So there's one thing we have to do to make sure this PR achieves its goal. ci.yml should use ubuntu 20.04 in the website step. 

https://github.com/cppalliance/mrdocs/blob/e460d86aceed8784d2cd3c7dd04daef51d74e3a8/.github/workflows/ci.yml#L323

Note that we should set the `container` and not `runs-on` to ensure we are running it in a clean container. If mrdocs builds in this environment, it means it's using its own libcxx because no other stdlib can be found in system paths.

We also have to remove steps this PR is supposed to make irrelevant:

We don't need to install clang because mrdocs is already supposed to have everything it needs:

https://github.com/cppalliance/mrdocs/blob/e460d86aceed8784d2cd3c7dd04daef51d74e3a8/.github/workflows/ci.yml#L327-L333

No need to install libstdc++-12 because mrdocs shouldn't be using the system stdlib:

https://github.com/cppalliance/mrdocs/blob/e460d86aceed8784d2cd3c7dd04daef51d74e3a8/.github/workflows/ci.yml#L339C20-L339C36

We might need to adjust it to install `build-essential` though. The reason is not related to mrdocs per se, but because cmake needs at least one compiler to run the configuration step. 

PS: The only reason we need some dummy compiler in the system is to run the cmake configuration step. It's technically possible to make MrDocs + CMake work without an existing compiler though: mrdocs would need to provide itself to cmake as the compiler when needed. That's more complex and outside the scope of this issue, but it's worth knowing.